From a432ebc8630c77e1da13b86b0405bb7570352f08 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Fri, 12 Jul 2013 14:10:58 +0300
Subject: [PATCH 03/14] net/lnet/lnet_main.c: #undefine M0_TRACE_SUBSYSTEM

+ net/lnet/ut/lnet_ut.c: Do not depend on lnet_main.c to #define
  M0_TRACE_SUBSYSTEM.
---
 net/lnet/lnet_main.c  | 3 ++-
 net/lnet/ut/lnet_ut.c | 7 ++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/net/lnet/lnet_main.c b/net/lnet/lnet_main.c
index cec7a5c..c4832e1 100644
--- a/net/lnet/lnet_main.c
+++ b/net/lnet/lnet_main.c
@@ -795,7 +795,7 @@ RPC Bulk Transfer Task Plan</a>
 #endif
 
 #define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_LNET
-#include "lib/trace.h"        /* M0_LOG and M0_ENTRY */
+#include "lib/trace.h"
 
 #include "lib/errno.h"
 #include "lib/misc.h"
@@ -1032,6 +1032,7 @@ M0_INTERNAL void m0_net_lnet_tm_set_debug(struct m0_net_transfer_mc *tm,
 M0_EXPORTED(m0_net_lnet_tm_set_debug);
 
 /** @} */ /* LNetDFS */
+#undef M0_TRACE_SUBSYSTEM
 
 /*
  *  Local variables:
diff --git a/net/lnet/ut/lnet_ut.c b/net/lnet/ut/lnet_ut.c
index 5657747..e0ed7e5 100644
--- a/net/lnet/ut/lnet_ut.c
+++ b/net/lnet/ut/lnet_ut.c
@@ -19,7 +19,10 @@
  * Original creation date: 1/4/2012
  */
 
-#include "net/lnet/lnet_main.c"
+#include "net/lnet/lnet_main.c" /* NB: #undefines M0_TRACE_SUBSYSTEM */
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_UT
+#include "lib/trace.h"
+
 #include "lib/arith.h" /* max64u */
 #include "lib/thread.h" /* m0_thread_self, m0_thread_handle_eq */
 #include "ut/ut.h"
@@ -2332,6 +2335,8 @@ const struct m0_test_suite m0_net_lnet_ut = {
 };
 M0_EXPORTED(m0_net_lnet_ut);
 
+#undef M0_TRACE_SUBSYSTEM
+
 /*
  *  Local variables:
  *  c-indentation-style: "K&R"
-- 
1.8.3.2

